comp/process/status/statusimpl/status.go

```diff
@@ -84,24 +86,33 @@ func (s statusProvider) getStatusInfo() map[string]interface{} {
 
 func (s statusProvider) populateStatus() map[string]interface{} {
 	status := make(map[string]interface{})
-	addressPort, err := config.GetProcessAPIAddressPort()
-	if err != nil {
-		status["error"] = fmt.Sprintf("%v", err.Error())
-		return status
-	}
 
 	c := client()
-	statusEndpoint := fmt.Sprintf("http://%s/agent/status", addressPort)
-	b, err := apiutil.DoGet(c, statusEndpoint, apiutil.CloseConnection)
+
+	var url string
+	if s.testServerURL != "" {
+		url = s.testServerURL
+	} else {
+		addressPort, err := config.GetProcessAPIAddressPort()
+		if err != nil {
+			status["error"] = fmt.Sprintf("%v", err.Error())
+			return status
+		}
+		url = fmt.Sprintf("http://%s/agent/status", addressPort)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Wonder if you could mock the `config.GetProcessAPIAddressPort` instead of setting the `testServerURL`. I'd rather use an interface and mock it in test.

Reply:
That is a totally valid point. I would like to do that in a separate PR, as `GetProcessAPIAddressPort` is used in multiple places and I don't want to change the focus of the fix in the PR 😄

Reply:
Oh I see config is global, ok then !